### PR TITLE
Add a '--cache' flag to virtctl for controlling disk cache mode

### PR DESCRIPTION
**Pull request by crazytaxii**
Signed-off-by: crazytaxii hua.feng@99cloud.net

**What this PR does / why we need it**:
Providing an option for controlling disk cache mode when using virtctl addvolume command to hotplug disk.

How to use: `virtctl addvolume --cache=writeback`

**Release note**:
```
Add a '--cache' flag to virtctl for controlling disk cache mode
```

### CODE REVIEW

**Review by maiqueb — CHANGES_REQUESTED** [review]
You'll need to add unit tests to check this PR does what you expect it to.

I think the correct place is in the corresponding [`virtctl` pkg](https://github.com/kubevirt/kubevirt/blob/main/pkg/virtctl/vm/vm_test.go).

File: pkg/virtctl/vm/vm.go [hunk 1]
```diff
@@ -71,6 +71,7 @@ const (
 	createClaimArg  = "create-claim"
 	storageClassArg = "storage-class"
 	accessModeArg   = "access-mode"
+  cacheArg        = "cache"
```
Comment (xpivarc) [inline pkg/virtctl/vm/vm.go:74]: Seems like broken formating.

File: tests/storage/hotplug.go [hunk 2]
```diff
@@ -122,6 +123,7 @@ var _ = SIGDescribe("Hotplug", func() {
 					},
 				},
 				Serial: volumeName,
+				Cache:  v1.DriverCache(cache),
```
Comment (xpivarc) [inline tests/storage/hotplug.go:126]: Do you want to add it here if `cache == ""`?

**Review by xpivarc — COMMENTED** [review]
Looks good. I missed a test verifying this change. Can we add one?

File: tests/storage/hotplug.go (hunk 2 above)
Comment (crazytaxii) [reply tests/storage/hotplug.go:126]: > Do you want to add it here if `cache == ""`?

I think `cache == ""` is illegal, an error message should be returned.

**Comment by crazytaxii** [comment]
> You'll need to add unit tests to check this PR does what you expect it to.
> 
> I think the correct place is in the corresponding [`virtctl` pkg](https://github.com/kubevirt/kubevirt/blob/main/pkg/virtctl/vm/vm_test.go).

How should I write the unit test? serialArg flag doesn't have such one in pkg/virtctl/vm/vm_test.go.

**Comment by awels** [comment]
I would be most interested in a functional test to ensure that the flag is actually passed to the disk when hotplugged into the VM.

**Comment by crazytaxii** [comment]
> Looks good. I missed a test verifying this change. Can we add one?

I have added one in tests/storage.

**Pull request merged**